### DISCUSSION TITLE
full height effect without h-screen?

### DISCUSSION THREAD

**Discussion by stackerito** [tailwindlabs/tailwindcss#5121] [Help]
I have the following snippet:
https://play.tailwindcss.com/xakrRwNxLv

As you can see I have a top panel for the sidebar which takes quite a lot of height (the orange panel)
The only way to fix it is if I set `h-screen` to the two `divs` below like so: https://play.tailwindcss.com/fFoDViAOKg

`h-full` doesn't work
But I'm afraid using `h-screen` is not the best solution in this case. what would be the optimal class for it?
(I want the top panel to occupy as less as possible and the content divs to occupy the rest - do I need to set fixed width to it?)

**Comment by Martleby** [comment]
I'm not clear what you want to achieve... but this might help: https://play.tailwindcss.com/Tgr8HIGvKK

**Reply by stackerito** [reply]
It's for a collapsible sidebar that stretches to the entire height of the screen
So I guess you have to give the wrapping div the `h-screen` class in order for the inner divs to be able to use `h-full` - because otherwise it will wrap the content?

**Reply by simonswiss** [reply]
Hey!

Yes, the parent that contains both elements (logo bit + sidebar) needs to have a given height (`h-screen` could work there) for the `h-full` to have any effect on the child element.

`h-full` will set `height: 100%`, which applies to the _container_ of what it's applied to.

Tip: it's always useful to make your Tailwind Play as minimal as possible. The ARIA attributes, transition component, SVG icons etc make it much harder to "parse" what's going on and what problem you're trying to illustrate/outline. 

Do you want your sidebar to stop the main content area to stop scrolling (kinda like a modal)? It seems so since you have a reduced opacity + fixed position on the main container).

You're likely to have quicker help if you do a "minimal" reproduction of your setup.
